### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by ExtraGravity**
This is similar to the `googletest` branch in Schulich Delta Desktop Telemetry, I have fixed a few of the comments from in there. `libgmock.a` is in the `../build/.lib/` folder. I also removed old tests, I plan to reimplement them in `googletest` in a later time, but as of right now there are more pressuring deadlines.

### CODE REVIEW

File: make-all.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+qmake
+make
```
Comment (JordanHeinrichs) [inline make-all.sh]: This should just be `make check` and then remove the ./../build/.tests/Tests.

File: make-all.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
```
Comment (JordanHeinrichs) [inline make-all.sh]: Add -e argument to this line. So if qmake fails then it won't try and make.

File: Tests/Tests.pro [hunk 3]
```diff
@@ -0,0 +1,21 @@
+TEMPLATE = app
+QT += testlib
+CONFIG += testcase
+
+LIBS += -L../../build/.lib -LC:\gtest\lib -lBusinessLayer -lCommunicationLayer -lDataLayer -lgmock
```
Comment (JordanHeinrichs) [inline Tests/Tests.pro]: As this is a Linux only application you can just remove -LC:\gtest\lib argument

File: make-all.sh (hunk 2 above)
Comment (ExtraGravity) [inline make-all.sh]: `#!/usr/bin/env bash -e` gives me `/usr/bin/env: sh -e: No such file or directory`.

**Pull request merged**